### ISSUE TITLE
[Bug/Error] Writing/Reading Cache from Network Drive

### ISSUE THREAD

**Issue by rikere** [rlguy/Blender-FLIP-Fluids-Beta#134]
I generally store all my blend files on a network server, which works for almost everything I do, but it looks like the FLIP Solver is having issues with it.

As near as I can tell, it starts up, succeeds at first, and manages to make it to frame 7-10 before it crashes out and puts an error in blender console. Strangely, the solver appears to actually keep running, even if I hit cancel. CPU usage stays high until I kill blender.exe directly.The connection to the server is pretty damned fast, full 1gb to a ZFS/Raid array, it's good enough to edit 4K footage without any real issues.

I saw there were 2 other issues/posts with possible network-drive related issues. Are there any specific test-cases/things I can run to help debug using networked storage with the FLIP addon?

I know I need to re-get the error, I forgot to copy it before just moving the blend to local storage.

**Comment by rlguy** [comment]
Thanks for letting me know! This sounds just like an issue another tester encountered while running a test case (#96). This issue was encountered when storing on a hard disk drive.

This bug may exist for all forms in storage, but is more likely to occur on HDD or on a network drive due to the slight delay. I have only tested the simulator on an SSD.

Capturing the error message will help with debugging this issue.

**Comment by rikere** [comment]
Apparently it's not putting it in the console, just a pop-up:

![image](https://user-images.githubusercontent.com/22123117/36620752-8e7ee8f4-18c2-11e8-88d8-c3e230f1c5e9.png)

Maybe allow it to use local-storage to write a cache, then it can copy it to remote storage?

**Comment by rikere** [comment]
Update: Tested playback from the network drive. That actually appeared to work pretty ok. Looks to be our only issue is going to be generating the simulation.

**Comment by rlguy** [comment]
Another tester reported a console error that seems like it is related to this issue:

```

Traceback (most recent call last):
  File "C:\Users\Ianed\AppData\Roaming\Blender Foundation\Blender\2.79\scripts\addons\bl_flip_fluids_test_suite\bake.py", line 1165, in bake
    __run_simulation(fluidsim, data, export_override, cache_directory, bakedata)
  File "C:\Users\Ianed\AppData\Roaming\Blender Foundation\Blender\2.79\scripts\addons\bl_flip_fluids_test_suite\bake.py", line 1080, in __run_simulation
    __write_simulation_output(domain, override_data, fluidsim, blender_frameno, cache_directory)
  File "C:\Users\Ianed\AppData\Roaming\Blender Foundation\Blender\2.79\scripts\addons\bl_flip_fluids_test_suite\bake.py", line 1041, in __write_simulation_output
    __write_autosave_data(domain_data, override_data, cache_directory, fluidsim, frameno)
  File "C:\Users\Ianed\AppData\Roaming\Blender Foundation\Blender\2.79\scripts\addons\bl_flip_fluids_test_suite\bake.py", line 1024, in __write_autosave_data
    shutil.rmtree(backup_autosave_dir)
  File "C:\Program Files\Blender Foundation\Blender\2.79\python\lib\shutil.py", line 494, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "C:\Program Files\Blender Foundation\Blender\2.79\python\lib\shutil.py", line 393, in _rmtree_unsafe
    onerror(os.rmdir, path, sys.exc_info())
  File "C:\Program Files\Blender Foundation\Blender\2.79\python\lib\shutil.py", line 391, in _rmtree_unsafe
    os.rmdir(path)
OSError: [WinError 145] The directory is not empty: 'C:\\Users\\Ianed\\Desktop\\flip\\cascading_water_feature_flip_fluid_cache\\savestates\\autosave_backup'
```

I have made some changes to attempt to fix this error.

I have attached the following file 'patch' and test this issue fix:

[bake.py.txt](https://github.com/rlguy/Blender-FLIP-Fluids-Beta/files/1769114/bake.py.txt)

You will need to rename the file to ```bake.py``` and replace the file that is located in your addon install directory. For example, my ```bake.py``` file is located in the following directory:

```
C:\Users\Ryan\AppData\Roaming\Blender Foundation\Blender\2.79\scripts\addons\bl_flip_fluids\
```

Please test if this if you are able, and let me know if this fixes the issue!

**Comment by rikere** [comment]
Patch didn't work. Produces error:

![image](https://user-images.githubusercontent.com/22123117/36880067-c4d73e98-1d94-11e8-8903-d487cdd73c37.png)

**Comment by rlguy** [comment]
Thanks for trying it out. It seems like Python isn't able to delete files with 100% success on the network. I'm trying different things to test getting around this. Does this version work?

[bake.py.txt](https://github.com/rlguy/Blender-FLIP-Fluids-Beta/files/1773509/bake.py.txt)

**Comment by rikere** [comment]
Will test tomorrow, is bedtime here.

I've noticed there is a sort of server-lag. Windows needs me to actively
refresh an explorer window in a server directory to see changes. I think
the OS might have some sort of cached file list to reduce load time from
the server. You might "delete" a file, but Windows won't believe it right
away.

If that patch doesn't work, I'd say overwrite the file if possible?

**Comment by rlguy** [comment]
This error should be fixed now. It will be included in the next version (**0.1.3b**) which should be available by March 31st.

If this error still occurs after the next version, let me know and I'll reopen this issue.
